apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift

```diff
@@ -29,26 +30,71 @@ public final class GraphQLQueryWatcher<Query: GraphQLQuery>: Cancellable, Apollo
   @Atomic private var fetching: WeakFetchTaskContainer = .init(nil, nil)
 
   @Atomic private var dependentKeys: Set<CacheKey>? = nil
+  private var subscriptions: [WatcherSubscription<AnySubscriber<CachePublisher.Output, Never>>] = []
 
-  /// Designated initializer
+  /// Convenience initializer
   ///
   /// - Parameters:
   ///   - client: The client protocol to pass in.
   ///   - query: The query to watch.
   ///   - context: [optional] A context that is being passed through the request chain. Defaults to `nil`.
   ///   - callbackQueue: The queue for the result handler. Defaults to the main queue.
   ///   - resultHandler: The result handler to call with changes.
-  public init(client: ApolloClientProtocol,
-              query: Query,
-              context: RequestContext? = nil,
-              callbackQueue: DispatchQueue = .main,
-              resultHandler: @escaping GraphQLResultHandler<Query.Data>) {
+  public convenience init(
+    client: ApolloClientProtocol,
+    query: Query,
+    context: RequestContext? = nil,
+    callbackQueue: DispatchQueue = .main,
+    resultHandler: @escaping GraphQLResultHandler<Query.Data>
+  ) {
+    self.init(
+      client: client,
+      query: query,
+      context: context,
+      callbackQueue: callbackQueue,
+      handler: resultHandler
+    )
+  }
+
+  /// Conenience initializer, intended for use with the watcher's `publisher` property.
+  /// - Parameters:
+  ///   - client: The client protocol to pass in.
+  ///   - query: The query to watch.
+  ///   - context: [optional] A context that is being passed through the request chain. Defaults to `nil`.
+  ///   - callbackQueue: The queue for the result handler. Defaults to the main queue.
+  public convenience init(
+    client: ApolloClientProtocol,
+    query: Query,
+    context: RequestContext? = nil,
+    callbackQueue: DispatchQueue = .main
+  ) {
+    self.init(
+      client: client,
+      query: query,
+      context: context,
+      callbackQueue: callbackQueue,
+      handler: nil
+    )
+  }
+
+  init(
+    client: ApolloClientProtocol,
+    query: Query,
+    context: RequestContext? = nil,
+    callbackQueue: DispatchQueue = .main,
+    handler: GraphQLResultHandler<Query.Data>?
```

Comment:
I would think the handler would need to be `@escaping` here also. What am I missing?